### ISSUE TITLE
Cannot save map file in TrenchBroom while Godot is open

### ISSUE THREAD

**Issue by GrauBlitz** [QodotPlugin/Qodot#53]
If i try to save my map inside TrenchBroom after i pressed "Build" inside Godot it just says "Cannot open file: <my map file>".

Maybe this is related to another script error #42 and therefore does not close the file after generating the map?

**Comment by DeerTears** [comment]
Can you post a screenshot of what you're trying to do? Build as in Build Map? Or Build as in the top-right `Build` button in the Mono editor?

Import issues are common enough in Godot but usually restarting the engine addresses them.

**Comment by Spooky309** [comment]
I'm getting the same issue on the latest master branch of Godot, after building the map once inside Godot, Trenchbroom won't let it save anymore:

![image](https://github.com/QodotPlugin/Qodot/assets/7798781/f2121efc-b3d6-4c9d-b5d2-e8d65509fce9)

There aren't any errors showing up in my Godot log, the only thing I can think of is that Qodot is (for whatever reason) failing to close its file handle after building as OP suggests.

**Comment by GrauBlitz** [comment]
Sorry for the late reply, I am building the map inside Godot ("Build" Button of your addon) and after that the file is locked (still opened inside Godot) so i cannot open it for writing (saving) again in TrenchBroom.

**Comment by ynot01** [comment]
Getting this also, a workaround is to reload the project and then save while Godot is still opening, still a bit hacky and tedious though

**Comment by ynot01** [comment]
> Edit: after adding a cheeky `file.Close();` into MapParser.cs:80, the problem has gone away, but allegedly FileAccess instances are refcounted so this shouldn't be required, probably an engine bug. 

Actually, in C# specifically, currently, RefCounted objects are not freed automatically, but checked periodically with a garbage collector.

From [Godot docs](https://docs.godotengine.org/en/stable/classes/class_refcounted.html):
```
Note: In C#, reference-counted objects will not be freed instantly after they are no longer in use. Instead, garbage collection will run periodically and will free reference-counted objects that are no longer in use. This means that unused ones will linger on for a while before being removed.
```

**Comment by MegadronA03** [comment]
same issue I fixed it, plugin forgots to close the file, gonna make merge request

**Comment by RhapsodyInGeek** [comment]
Resolved by #62
